tests/framework/extensions/clusters/kubernetesversions/available.go

```diff
@@ -11,108 +11,185 @@ import (
 	"github.com/rancher/rancher/tests/framework/clients/rancher"
 	v3 "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
 	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/sirupsen/logrus"
 )
 
 // ListRKE1AvailableVersions is a function to list and return only available RKE1 versions for a specific cluster.
 func ListRKE1AvailableVersions(client *rancher.Client, cluster *v3.Cluster) (availableVersions []string, err error) {
-	allAvailableVersions, err := ListRKE1AllVersions(client)
+	var allAvailableVersions []*semver.Version
+	allRKE1Versions, err := ListRKE1AllVersions(client)
 	if err != nil {
 		return
 	}
 
-	for i, version := range allAvailableVersions {
-		if strings.Contains(version, cluster.RancherKubernetesEngineConfig.Version) {
-			availableVersions = allAvailableVersions[i+1:]
-			break
+	for _, v := range allRKE1Versions {
+		rkeVersion, err := semver.NewVersion(strings.TrimPrefix(v, "v"))
+		if err != nil {
+			logrus.Errorf("couldn't turn %v to sem ver", rkeVersion)
```

Comment:
extreme nit; should it be `semver` or `sem ver` fine?

Reply:
I agree here, not a big deal though

Reply:
Updated as: `logrus.Errorf("couldn't turn %v to a semantic version", rkeVersion)`.  `semver` is a package name, so I changed them as `a semantic version`.